app/javascript/dashboard/routes/dashboard/settings/attributes/AddAttributes.vue

```diff
@@ -0,0 +1,194 @@
+        </div>
+        <div v-if="hasValue" class="medium-12 columns">
+          <label>
+            {{ $t('ATTRIBUTES_MGMT.ADD.FORM.KEY.LABEL') }}
+            <i class="ion-help" />
+          </label>
+          <p class="key-value text-truncate">
+            {{ convertToSlug }}
+          </p>
+        </div>
+        <div class="modal-footer">
+          <div class="medium-12 columns">
+            <woot-submit-button
+              :disabled="
+                $v.displayName.$invalid ||
+                  $v.description.$invalid ||
+                  uiFlags.isCreating
+              "
+              :button-text="$t('ATTRIBUTES_MGMT.ADD.SUBMIT')"
+            />
+            <button class="button clear" @click.prevent="onClose">
+              {{ $t('ATTRIBUTES_MGMT.ADD.CANCEL_BUTTON_TEXT') }}
+            </button>
+          </div>
+        </div>
+      </form>
+    </div>
+  </woot-modal>
+</template>
+
+<script>
+import { required, minLength } from 'vuelidate/lib/validators';
+import { mapGetters } from 'vuex';
+
+export default {
+  props: {
+    onClose: {
+      type: Function,
+      default: () => {},
+    },
+  },
+  data() {
+    return {
+      displayName: '',
+      description: '',
+      attributeModel: 0,
+      attributeType: 0,
+      models: [
+        {
+          id: 0,
+          option: 'Conversation',
+        },
+        {
+          id: 1,
+          option: 'Contact',
+        },
+      ],
+      types: [
+        {
+          id: 0,
+          option: 'Text',
+        },
+        {
+          id: 1,
+          option: 'Number',
+        },
+        {
+          id: 2,
+          option: 'Currency',
+        },
+        {
+          id: 3,
+          option: 'Percent',
+        },
+        {
+          id: 4,
+          option: 'Link',
+        },
+      ],
+      show: true,
+    };
+  },
+  computed: {
+    ...mapGetters({
+      uiFlags: 'getUIFlags',
+    }),
+    convertToSlug() {
+      return this.displayName
+        .toLowerCase()
+        .replace(/[^\w ]+/g, '')
+        .replace(/ +/g, '_');
+    },
+    hasValue() {
+      return this.displayName.length >= 1;
+    },
+  },
+  validations: {
+    displayName: {
+      required,
+      minLength: minLength(1),
+    },
+    description: {
+      required,
+    },
+    attributeModel: {
+      required,
+    },
+    attributeType: {
+      required,
+    },
+  },
+
+  methods: {
+    showAlert(message) {
+      bus.$emit('newToastMessage', message);
+    },
+    async addAttributes() {
+      try {
+        await this.$store.dispatch('attributes/create', {
+          attribute_display_name: this.displayName,
+          attribute_description: this.description,
+          attribute_model: this.attributeModel,
+          attribute_display_type: this.attributeType,
+          attribute_key: this.convertToSlug,
+        });
+        this.showAlert(this.$t('ATTRIBUTES_MGMT.ADD.API.SUCCESS_MESSAGE'));
+        this.onClose();
+      } catch (error) {
```

Comment:
Check any error message that exists in the `error` object otherways show the default error message.